src/algorithm/centroid.rs

```diff
@@ -0,0 +1,87 @@
+pub use types::{Coordinate, Point, LineString};
+
+pub trait Centroid {
+	/// Calculation the centroid, see: https://en.wikipedia.org/wiki/Centroid
+    fn centroid(&self) -> Result<Point, &'static str>;
```

Comment:
As the error of the centroid calculation seems trivial, what about just returning an Option?